ops_utils/tdr_utils/tdr_api_utils.py

```diff
@@ -180,6 +182,58 @@ def delete_files(
             check_interval=check_interval
         ).run()
 
+    def _delete_snapshots(self, dataset_id: str, file_ids: set[str]) -> None:
+        """Delete snapshots that reference any of the provided file IDs."""
+        snapshots_resp = self.get_dataset_snapshots(dataset_id=dataset_id)
+        snapshot_items = snapshots_resp.json().get('items', [])
+        snapshots_to_delete = []
+        logging.info(
+            "Checking %d snapshots for references",
+            len(snapshot_items),
+        )
+        for snap in snapshot_items:
+            snap_id = snap.get('id')
+            if not snap_id:
+                continue
+            snap_files = self.get_files_from_snapshot(snapshot_id=snap_id)
+            snap_file_ids = {
+                fd.get('fileId') for fd in snap_files if fd.get('fileId')
+            }
+            # Use set intersection to check for any matching file IDs
+            if snap_file_ids & file_ids:
+                snapshots_to_delete.append(snap_id)
+        if snapshots_to_delete:
+            logging.info(
+                f"{self.dry_run_msg()}Deleting {len(snapshots_to_delete)} snapshots that reference "
+                "target files")
+            if not self.dry_run:
+                for snap_id in snapshots_to_delete:
+                    job_id = self.delete_snapshot(snap_id).json()['id']
```

Comment:
Could this use the `delete_snapshots` method so it can submit multiple jobs now?